src/main/webapp/app/tasks/local-task.service.ts

```diff
@@ -1,43 +1,62 @@
-import { Injectable } from '@angular/core';
-import { Observable, of } from 'rxjs';
-import { v4 as uuid } from 'uuid';
+import { Injectable } from '@angular/core'
+import { Observable, of } from 'rxjs'
+import { v4 as uuid } from 'uuid'
 
-import { Task } from './task';
-import { TaskService } from './task.service';
+import { Task } from './task'
+import { TaskService } from './task.service'
 
 @Injectable()
 export class LocalTaskService implements TaskService {
-
-  private static readonly STORAGE_KEY: string = 'tiny.tasks';
+  private static readonly STORAGE_KEY: string = 'tiny.tasks'
 
   getAll(): Observable<Task[]> {
-    return of(this.readTasks());
+    return of(this.readTasks())
   }
 
   create(name: string): Observable<Task> {
-    const tasks = this.readTasks();
-    const task = {id: uuid(), name};
-    tasks.push(task);
-    this.writeTasks(tasks);
-    return of(task);
+    const tasks = this.readTasks()
+    const task = { id: uuid(), name, status }
+    tasks.push(task)
+    this.writeTasks(tasks)
+    return of(task)
   }
 
   delete(id: string): Observable<void> {
-    const tasks = this.readTasks();
-    const index = tasks.findIndex(task => task.id === id);
+    const tasks = this.readTasks()
+    const index = tasks.findIndex(task => task.id === id)
+    if (index !== -1) {
+      tasks.splice(index, 1)
+      this.writeTasks(tasks)
+    }
+    return of(null)
+  }
+
+  setIsDone(id: string): Observable<void> {
+    const tasks = this.readTasks()
+    tasks.forEach(task =>
+      task.id === id ? (task.status = 'done') : (task.status = 'to do')
+    )
+    // here or in the task-list.component.ts I would sort the tasks by their status
```

Comment:
right track, but how would you do it?

Reply:
I thought about somethink like  
```
const a = tasks.status = 'done'
const b = tasks.status = 'to do'
tasks.sort(a, b){return a - b}
```
But this don't work. So I think it should be easier with a boolean. Because then it's easier to use the localeCompare(). Something like this  `.sort((x,y) => x.task.status === y.task.status ? 0 : x ? 1 : -1))`

What would you prefer?